grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ConceptImpl.java

```diff
@@ -57,6 +57,7 @@
         Optional<Shard> shard = vertex().tx().factory().buildShard(shardVertex);
         return shard.orElseThrow(() -> GraknTxOperationException.missingShard(getId()));
     });
+    private final Cache<Long> shardCount = new Cache(Cacheable.number(), () -> shards().count());
```

Comment:
waiiiit... why isn't this in `TypeImpl`?

Reply:
Because `technically` all concept can have shards and eventually will. Relationship sharding for example.

Reply:
👍👍👍  looking forwards to instance sharding